### PR TITLE
1248 - Datagrid suppress duplicate filtered events on datepicker and timepicker

### LINKED ISSUE #1248: IdsDatagrid: Column header date filter button is displayed incorrectly and filtered event is triggered multiple times
**Describe the bug**
The datepicker button is displayed on the right side when the column headeralign is right, it should be left.
image
Expected:
image

**To Reproduce**

Headeralign issue:
1. open enterprise-wc
2. update disable-client-filter.ts, add a date column with headeralign: right 
`
  columns.push({
    id: 'publishDate',
    name: 'Pub. Date',
    field: 'publishDate',
    width: 180,
    filterType: dataGrid.filters.date,
    formatter: dataGrid.formatters.date,
    headerAlign: `right`
  });
`
3. run and open disable-client-filter.html
4. Error: see Publish date column filter, the button is on the wrong side. Should be on the left

Event filtered issue 
steps 1 to 3 are same as above
4. Select a date on the date filter then press enter
5. Error: see console log, filtered event was triggered multiple times

**Expected behavior**
1) Date picker button should be on the right side of the filter for headeralign right
2) 'filtered' event should only execute once when pressing enter or selecting from datepicker header filter

**Version**

- ids enterprise wc 1.0.0-beta.8

**Screenshots**

**Platform**
 - Infor Application/Team Name: M3 UX
 - OS Version: Windows 11
 - Browser Name: [chrome
 - Browser Version: 112.0.5615.121 (Official Build) (64-bit)

**Additional context**

### PR DESCRIPTION
**Pull request by clayinfor**
Fix to suppress extraneous 'filtered' events, where on datagrid, selecting a date filter or a time filter would fire multiple `filtered` events.

**Explain the details for making this change. What existing problem does the pull request solve?**

**Related github/jira issue (required)**:

Closes #1248 

**Steps necessary to review your pull request (required)**:

1. Check out branch and run: `nvm use && npm run start`
2. Open http://localhost:4300/ids-data-grid/disable-client-filter.html
3. Filter the datagrid using the date-filter column, and ensure only 1 `filtered` event is fired in dev console.
4. Filter the datagrid using the time-filter column, and ensure only 1 `filtered` event is fired in dev console.
5. Run tests to ensure they all pass: `npx jest -- data-grid-func-test`

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

File: src/components/ids-data-grid/ids-data-grid-filters.ts [hunk 1]
```diff
@@ -902,14 +902,16 @@ export default class IdsDataGridFilters {
 
     // Capture 'dayselected' events from IdsDatePickerPopup
     this.root.onEvent(`dayselected.${this.#id()}`, this.root.wrapper, (e: any) => {
+      // this will trigger the above change event which also calls applyFilter().
       e.target.value = e.detail.value;
-      this.applyFilter();
+      // this.applyFilter(); // not needed, see above comment
```
Comment (tmcconechy) [inline src/components/ids-data-grid/ids-data-grid-filters.ts:907]: Can remove if not needed

**Review by tmcconechy — APPROVED** [review]
Looks good to me ~ just suggest removing commented out lines vs keeping them in.. (only a suggestion)

**Review by tmcconechy — APPROVED** [review]
Looks good to me ~ just suggest removing commented out lines vs keeping them in.. (only a suggestion)

**Comment by clayinfor** [comment]
Thanks @tmcconechy , I've removed commented-out code as suggested :)

**Review by tmcconechy — DISMISSED** [review]

**Review by deep7102 — COMMENTED** [review]
Looks good, just tests need to adjust

**Comment by clayinfor** [comment]
Thanks @deep7102 !  The failed tests are IdsDropdown related @tmcconechy.

**Pull request merged**